### ISSUE TITLE
Build failed with the latest nightly rustc

### ISSUE THREAD

**Issue by zonyitoo** [contain-rs/linked-hash-map#31]
```
Compiling linked-hash-map v0.0.6
/root/.cargo/registry/src/github.com-121aea75f9ef2ce2/linked-hash-map-0.0.6/src/lib.rs:31:1: 31:28 error: #[feature] may not be used on the stable release channel
/root/.cargo/registry/src/github.com-121aea75f9ef2ce2/linked-hash-map-0.0.6/src/lib.rs:31 #![feature(hashmap_hasher)]
^~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/.cargo/registry/src/github.com-121aea75f9ef2ce2/linked-hash-map-0.0.6/src/lib.rs:32:1: 32:21 error: #[feature] may not be used on the stable release channel
/root/.cargo/registry/src/github.com-121aea75f9ef2ce2/linked-hash-map-0.0.6/src/lib.rs:32 #![feature(box_raw)]
^~~~~~~~~~~~~~~~~~~~
/root/.cargo/registry/src/github.com-121aea75f9ef2ce2/linked-hash-map-0.0.6/src/lib.rs:33:1: 33:24 error: #[feature] may not be used on the stable release channel
/root/.cargo/registry/src/github.com-121aea75f9ef2ce2/linked-hash-map-0.0.6/src/lib.rs:33 #![feature(iter_order)]
^~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to 3 previous errors
```

**Comment by apasel422** [comment]
Can you post the output of running `rustc --version`?

**Comment by zonyitoo** [comment]
I just upgraded the latest rustc nightly and got these messages

```
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:664:38: 664:53 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:664         self.len() == other.len() && iter::order::eq(self.iter(), other.iter())
                                                                                                                                                    ^~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:664:38: 664:53 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:668:38: 668:53 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:668         self.len() != other.len() || iter::order::ne(self.iter(), other.iter())
                                                                                                                                                    ^~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:668:38: 668:53 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:676:9: 676:33 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:676         iter::order::partial_cmp(self.iter(), other.iter())
                                                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:676:9: 676:33 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:680:9: 680:24 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:680         iter::order::lt(self.iter(), other.iter())
                                                                                                                       ^~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:680:9: 680:24 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:684:9: 684:24 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:684         iter::order::le(self.iter(), other.iter())
                                                                                                                       ^~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:684:9: 684:24 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:688:9: 688:24 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:688         iter::order::ge(self.iter(), other.iter())
                                                                                                                       ^~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:688:9: 688:24 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:692:9: 692:24 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:692         iter::order::gt(self.iter(), other.iter())
                                                                                                                       ^~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:692:9: 692:24 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:698:9: 698:25 error: use of unstable library feature 'iter_order_deprecated': needs review and revision (see issue #27737)
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:698         iter::order::cmp(self.iter(), other.iter())
                                                                                                                       ^~~~~~~~~~~~~~~~
/Users/zonyitoo/.multirust/cargo/registry/src/github.com-0a35038f75765ae4/linked-hash-map-0.0.6/src/lib.rs:698:9: 698:25 help: add #![feature(iter_order_deprecated)] to the crate attributes to enable
error: aborting due to 8 previous errors
Build failed, waiting for other jobs to finish...
Could not compile `linked-hash-map`.
```

```
$ rustc --version
rustc 1.6.0-nightly (1a2eaffb6 2015-10-31)
```

**Comment by apasel422** [comment]
Fixed in version 0.0.7.
